nameko/messaging.py

```diff
@@ -350,8 +350,11 @@ def on_iteration(self):
             self.should_stop = True
 
     def on_connection_error(self, exc, interval):
-        _log.warn('broker connection error: {}. '
-                  'Retrying in {} seconds.'.format(exc, interval))
+        _log.warn(
+            "Error connecting to broker at {} ({}).\n"
+            "Is the broker running? "
+            "Do you have permission to access to the vhost?\n"
+            "Retrying in {} seconds.".format(self.amqp_uri, exc, interval))
```

Comment:
Re https://github.com/onefinestay/nameko/issues/236, I think this is as good as we can do. Kombu doesn't give you anything more specific than "socket closed"